### ISSUE TITLE
t.web2.static.File does not hand index files off to a processor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2963]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| PenguinOfDoom reported|
|-|-|
|Trac ID|trac#2963|
|Type|defect|
|Created|2007-12-21 04:35:05Z|
|Branch|https://github.com/twisted/twisted/tree/web2-processors-2963|

File.indexNames is consulted at render() time, but processors are checked before then, in locateChild

<details><summary>Searchable metadata</summary>

```
trac-id__2963 2963
type__defect defect
reporter__PenguinOfDoom PenguinOfDoom
priority__normal normal
milestone__ 
branch__branches_web2_processors_2963 branches/web2-processors-2963
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__web2 web2
keywords__ 
time__1198211705000000 1198211705000000
changetime__1297622062000000 1297622062000000
version__None None
owner__ 
cc__dreid cc__pahan
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|PenguinOfDoom commented|
|-|-|

This was broken in r15523

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|pahan commented|
|-|-|

(In [22104]) Branching to 'web2-processors-2963'

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|PenguinOfDoom set owner to @dreid|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|PenguinOfDoom set owner to @wsanchez|
|-|-|

Whoops, looks like this breaks DAV. Maybe DAV shouldn't reuse so much of static.File code?

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_5"></a>|@wsanchez set owner to @dreid|
|-|-|

DAV?  What DAV?

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_6"></a>|@wsanchez commented|
|-|-|

Really, all the processors crap should be in a subclass of `File`, or in a MixIn.  One should be able to get a `File` that just does `File` things, and processors should be add-on functionality.

We have a tendency in `web2` to overload functionality on classes.  `Resource` is another example, which is why `File`, for example, doesn't subclass it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Having a really simple, bare-bones `File` sounds like a pretty great thing.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

Not going to fix this in web2 (see [#4821](https://github.com/twisted/twisted/issues/4821)).  `twisted.web.static.File` does hand index files off to processors, so not retargeting this at the web component.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
